### PR TITLE
337: Reapply stress test changes into api-refactor

### LINKED ISSUE #337: Document limits and discoveries from stress tests
We need a way to communicate our service limits and findings to our users. A new dedicated page in our official docs is probably the best way to accomplish this.

Spin-off from https://github.com/quintilesims/layer0/pull/333.

### PR DESCRIPTION
**Pull request by jparsons04**
**What does this pull request do?**
Issue #232 had made some changes to reconfigure stress tests to use benchmarking. Those changes were merged into the develop branch, and was overwritten after the release of v0.10.4.

This puts the stress test changes back into api-refactor

**How should this be tested?**
`make benchmark` from the root of the project or within the `tests/stress` directory

**Checklist**
~- [ ] Unit tests~
~- [ ] Smoke tests (if applicable)~
~- [ ] System tests (if applicable)~
~- [ ] Documentation (if applicable)~

closes #337

### CODE REVIEW

**Comment by jparsons04** [comment]
Gist for the first attempt at running the stress tests: https://gist.github.com/jparsons04/eb17289d521c8082cf4abf9ec2d246c0

Perhaps we should consider recalibrating the limits within the stress test.

**Comment by diemonster** [comment]
It seems like getting stress tests to "work" while we rip out so much code is going to be something we'll need to readdress regardless. I would say we merge this for now but examine it again after we get in a more stable spot (i.e. don't close #337 yet).

File: common/testutils/functions.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package testutils
+
+import (
+	"reflect"
+	"time"
+)
+
+func AssertEqual(t Tester, result, expected interface{}) {
```
Comment (zpatrick) [inline common/testutils/functions.go]: Any `assert` functions should be removed, we have the `testify/assert` package for this.

File: common/testutils/tester_interface.go [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package testutils
```
Comment (zpatrick) [inline common/testutils/tester_interface.go]: file should be `tester.go`

File: tests/clients/layer0_test_client.go [hunk 3]
```diff
@@ -23,93 +21,169 @@ func NewLayer0TestClient(t *testing.T, endpoint, token string) *Layer0TestClient
 	}
 }
 
-func (l *Layer0TestClient) jobHelper(fn func() (string, error)) string {
-	jobID, err := fn()
+func (l *Layer0TestClient) CreateTask(taskName, environmentID, deployID string, copies int, overrides []models.ContainerOverride) string {
```
Comment (zpatrick) [inline tests/clients/layer0_test_client.go]: This is not the established pattern: we pass in request models now. The test client has already been updated to reflect this; any additional functionality should adhere to those patterns.

**Pull request closed without merging**